pkg/maps/metricsmap/metricsmap.go

```diff
@@ -137,47 +137,134 @@ func (vs Values) Bytes() uint64 {
 	return b
 }
 
-func updateMetric(getCounter func() (prometheus.Counter, error), newValue float64) {
-	counter, err := getCounter()
+// metricsMapCollector implements Prometheus Collector interface
+type metricsmapCollector struct {
+	droppedCountDesc *prometheus.Desc
+	droppedByteDesc  *prometheus.Desc
+	forwardCountDesc *prometheus.Desc
+	forwardByteDesc  *prometheus.Desc
+
+	// eBPF code seems to expose multiple reasons for forwarded metrics
+	// as opposed to what is stated in bpf/lib/metrics.h comments.
+	// IterateWithCallback will iterate through BPF map for each reason and direction.
```

Comment:
Verb tense nit
```suggestion
	// IterateWithCallback iterates through BPF map for each reason and direction.
```

Reply:
@zacharysarah I'm really sorry for making confusion, but all changes you have proposed are not from my PR. I tried to amend the unsigned commit as suggested by @dylandreimerink and I've pushed someone else's commits to this branch by mistake.
The only thing that you have commented that is part of this PR is regarding the wording in metricsmap.go, and I have fixed that in [#d1988ae](https://github.com/cilium/cilium/pull/27370/commits/d1988ae903d5b7424af189c6bce16bf253c275d6)

Reply:
@carnerito
> all changes you have proposed are not from my PR
And yet your changes touch them, creating an opportunity to make docs better. Otherwise--when will docs ever improve?

I'm disappointed in your choice to refuse uncomplicated improvements, but I'll approve regardless.

Reply:
It looks to me like there was a misunderstanding here in the way that the changes got pushed into GitHub and the way that GitHub requested reviewers for these changes. From what I can tell, @carnerito never intended to make changes to these files in the PR, but the way that the PR was pushed into GitHub caused GitHub to pull @zacharysarah in for review.

In future it might be worth doing a quick double-check of the PR before marking it "ready for review" to make sure that GitHub presents the set of changes that you expected to present, so that GitHub only pulls in the codeowners for the areas that are being changed.

At this point it looks like @carnerito has been able to resolve this issue, maybe with a fresh push of the branch, and none of these comments are relevant to the changes in the PR any more. Either way it looks like this has been resolved in the end. @carnerito apologies for the confusion around this, thanks again for contributing these changes to Cilium :-)

Reply:
@joestringer Thanks for the explanation. @carnerito Apologies for pushing back so strongly; I see the underlying issue now. Thanks for your contributions!